packages/sdk/src/search/useInfiniteSearchState.ts

```diff
@@ -16,24 +16,56 @@ type Action =
       payload: number
     }
 
+// Save the array containing loaded pages before navigating away from the PLP
+function setPagesSessionStorage(pages: number[]) {
+  try {
+    // Uses the key to identify a product gallery page
+    const stateKey = window.history.state?.key
+    if (!stateKey) {
+      return
+    }
+    const storageKey = `__fs_gallery_page_${stateKey}`
+
+    sessionStorage.setItem(storageKey, JSON.stringify(pages))
+  } catch (error) {
+    return
+  }
+}
+
+function retrievePagesFromSessionStorage(): number[] | null {
```

Comment:
nit: what about `getPagesFromSessionStorage`, instead?

Reply:
I don't think this is nitpicking, actually. We expect a `get` after we see a `set`.